### ISSUE TITLE
use src-layout

### ISSUE THREAD

**Issue by nstarman** [astropy/quantity-2.0#7]
Let's use the modern package layout before we're too entrenched!

There are many benefits:
1. .egg files are not dumped in the top namespace
2. package build tools default settings are to find packages in src directories. We *currently* need less config and guarantee we *will* need less config no matter what build system we use over time.
3. The `src` layout also makes having support packages much easier.

As potential examples

```
src/
    quantity/*  # core library
    quantity_mpl/  # installed if extra [mpl]. This manages matplotlib interop
    quantity_xarray/  # manages xarray interop

```

**Comment by mhvk** [comment]
Despite having destroyed the `src` layout when I used the openastronomy template, I am fine with using it for this new package -- I still dislike that `src` no longer is  something that needs to be put in a compiler, but not so much that I cannot see the advantages (and perhaps we're going there anyway with `mypyc`...).

**Comment by mhvk** [comment]
Anyway, with a bit of further thought, since I don't really care, let's just move to the `src` layout.

**Comment by nstarman** [comment]
Closed by #12
